lib/caching-precompiler.js

```
@@ -48,8 +49,32 @@ CachingPrecompiler.prototype._init = function () {
 CachingPrecompiler.prototype._transform = function (code, filePath, hash) {
 	code = code.toString();
 
-	var options = babelConfigHelper.build(this.babelConfig, this.powerAssert, filePath, code);
-	var result = this.babel.transform(code, options);
+	var fileExt = path.extname(filePath).replace(/^\./, '');
+	var isCustomExt = this.extensions.filter(function (ext) {
+		return (ext !== 'js' && ext === fileExt);
+	}).length > 0;
+
+	var result;
+
+	if (isCustomExt) {
+		switch (fileExt) {
+			case 'ts':
+				var extTs = require('./ext/ts');
+				result = extTs.compile(code, filePath);
+				break;
+
+			// here for other exts in the future
+
+			default:
+				// not supported(yet)
```

It should throw an useful error here.